### PR TITLE
Dev/ft/ring 28845 use uuids

### PR DESCRIPTION
**Pull request by maximelubin**
Currently targeting ecn branch until merged, eventually should target master directly.
hdclient now expects on startup a new parameter 'uuidmapping' containing either:
* a valid JS object (for test and script)
* a string pointing to a JSON file (evenutally available as a Kubernetes ConfigMap)

### CODE REVIEW

**Review by jgagnaire — APPROVED** [review]

File: lib/hdclient.js [hunk 1]
```diff
@@ -155,7 +171,8 @@ class HyperdriveClient {
               callback, size, inputStream,
               httpAgent: this.httpAgent,
               errorAgent: this.errorAgent,
-              requestTimeoutMs: this.options.requestTimeoutMs });
+              requestTimeoutMs: this.options.requestTimeoutMs,
+              uuidmapping: this.options.uuidmapping });
```
Comment (xroche-scality) [inline lib/hdclient.js]: Pedantic minor:  diff-friendly ?

              requestTimeoutMs: this.options.requestTimeoutMs,
              uuidmapping: this.options.uuidmapping,
              });

**Review by xroche-scality — APPROVED** [review]

File: lib/hdclient.js (hunk 1 above)
Comment (maximelubin) [reply lib/hdclient.js]: done

File: lib/hdclient.js [hunk 2]
```diff
@@ -105,6 +107,20 @@ class HyperdriveClient {
...
+     * Use passed object or load JSON file
+     *
+     * @param {String|Object} uuidmapping - Mapping UUID on host:port
+     * @return {undefined}
+     */
+    setupUuidMapping(uuidmapping) {
+        if (typeof(uuidmapping) === 'string') {
+            const loaded = fs.readFileSync(uuidmapping);
+            this.options.uuidmapping = JSON.parse(loaded);
+        }
```
Comment (lunatikub) [inline lib/hdclient.js:121]: no error management if uuidmapping is not of type string ?
assert ?

**Review by lunatikub — APPROVED** [review]

File: lib/hdclient.js (hunk 2 above)
Comment (maximelubin) [reply lib/hdclient.js:121]: If it is not a stirng, I assume this.options.uuidmapping already contains a valid JS object doing the ampping. Not super proud of this code though

Comment (lunatikub) [reply lib/hdclient.js:121]: Okay :+1: 
So a comment can be good :)

**Pull request merged**